receiver/simpleprometheusreceiver/examples/federation/prom-counter/main.go

```diff
@@ -63,7 +62,9 @@ func main() {
 	meter := initMeter()
 	// logging
 	logger, _ := zap.NewProduction()
-	defer logger.Sync()
+	defer func() {
```

Comment:
Why do you need to wrap this in a func?

Reply:
No special reason, just follow other issues' way

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12592/files#diff-d4418701fb60e9de7b034cb0351c95e197998f612cd322e2d4272c342055f1f7R111-R113

Reply:
I believe there is no other way to satisfy errcheck linter